src/server/service/match.ts

```diff
@@ -178,22 +209,22 @@ export class Match {
       }
 
       if (!player.isBot) {
-        // TODO: Implement player stats functionality
-        //const playerStats = getPlayerStats(player.id);
-
         // Check achievements
         const achievementPoints = Object.entries(MATCH_ACHIEVEMENTS)
-          .map(([id, achievement]) => {
-            const pointsEarned = achievement.calculate({
+          .filter(([_, achievement]) => {
+            return achievement.calculate({
               player,
               messages: this._messages,
+              botsBusted,
+              otherPlayers,
+              playerHistory: this._playerPreviousMatches[player.userId],
             });
-            return { id, points: pointsEarned };
           })
-          .filter((achievement) => achievement.points > 0)
-          .reduce((totalPoints, achievement) => {
-            player.achievements.push(achievement);
-            return (totalPoints += achievement.points);
+          .reduce((totalPoints, [id, _]) => {
+            const achievementId = achievementIdSchema.safeParse(id);
```

Comment:
if this is an ID it shouldn't have a `success` prop. Is it then a full achievement object?

Reply:
I don't know what you mean by this..

Reply:
I do not know what you mean by this?